client/extensions/woocommerce/app/dashboard/index.js

```diff
@@ -20,55 +23,68 @@ class Dashboard extends Component {
 		className: PropTypes.string,
 	};
 
-	onStoreSetupFinished = () => {
-		// TODO - save that setup has been finished to the store's state on WPCOM
-		// TODO - is there a way to set an option on the store site?
+	componentDidMount = () => {
+		const { selectedSite } = this.props;
+
+		if ( selectedSite && selectedSite.ID ) {
+			this.props.fetchSetupChoices( selectedSite.ID );
+		}
 	}
 
-	renderStoreSetup = () => {
+	componentWillReceiveProps = ( newProps ) => {
 		const { selectedSite } = this.props;
-		return (
-			<Setup
-				onFinished={ this.onStoreSetupFinished }
-				site={ selectedSite }
-			/>
-		);
-	}
 
-	renderStoreManagement = () => {
-		const { translate } = this.props;
+		const newSiteID = newProps.selectedSite && newProps.selectedSite.ID || null;
+		const oldSiteID = selectedSite && selectedSite.ID || null;
 
-		return (
-			<Card>
-				<p>
-					{ translate( 'This is the start of something great!' ) }
-				</p>
-				<p>
-					{ translate( 'This will be the home for your WooCommerce Store integration with WordPress.com.' ) }
-				</p>
-			</Card>
-		);
+		if ( oldSiteID !== newSiteID ) {
+			this.props.fetchSetupChoices( newSiteID );
+		}
+	}
+
+	onStoreSetupFinished = () => {
+		// TODO - save that setup has been finished to the store's state on WPCOM
 	}
 
 	render = () => {
-		const { storeSetupCompleted } = this.props;
+		const { finishedInitialSetup, loading, selectedSite } = this.props;
+
+		if ( loading || ! selectedSite ) {
+			// TODO have a placholder/loading view instead
+			return null;
+		}
 
 		return (
 			<Main className={ classNames( 'dashboard', this.props.className ) }>
-				{ storeSetupCompleted ? this.renderStoreManagement() : this.renderStoreSetup() }
+				{
+					finishedInitialSetup && <Manage site={ selectedSite } /> ||
+					<Setup onFinished={ this.onStoreSetupFinished } site={ selectedSite } />
+				}
 			</Main>
 		);
 	}
 
 }
 
 function mapStateToProps( state ) {
-	// TODO - figure out from state if setup has been completed for this store yet
+	const finishedInitialSetup = getFinishedInitialSetup( state );
+	const loading = areSetupChoicesLoading( state );
+	const selectedSite = getSelectedSite( state );
```

Comment:
Thank you! Good catch. Fixed in https://github.com/Automattic/wp-calypso/pull/14339/commits/5512a8530a3394de9a0cedd0beb566762edddebc